git-clone/entrypoint.sh

```diff
@@ -3,20 +3,35 @@
 # The entrypoint removes the previous mount-path and does a fresh
 # checkout of the repository. It also initializes git lfs and sets
 # the proper file permissions.
-set -x
 
 if [ "$LFS_AUTO_FETCH" = 1 ]; then
   LFS_SKIP_SMUDGE="";
 else
   LFS_SKIP_SMUDGE="--skip-smudge";
 fi
 
+# clear path
 rm -rf ${MOUNT_PATH}/*
 (rm -rf ${MOUNT_PATH}/.* || true)
+
+# set up git defaults
 git config --system push.default simple
-git lfs install $LFS_SKIP_SMUDGE --system
-git clone $REPOSITORY ${MOUNT_PATH}
+
+# extract the GitLab host and path
+pat='^(http[s]?:\/\/)([^\/]+)\/?([a-zA-Z0-9_\/\-]+)?$'
```

Comment:
```suggestion
pat='^(http[s]?:\/\/)([^\/]+)\/?([a-zA-Z0-9_\/\-]+?)$'
```
The bash version in alpine linux seems to give preference to greedy matches closer to the end of the string, while the version in ubuntu or on mac is "greedier" at the beginning. By making the path regex lazy while the hostname regex remains greedy, the problem is fixed.